internal/controllers/rest_controller.go

```diff
@@ -172,11 +183,99 @@ func (rc RESTController) Logout(w http.ResponseWriter, r *http.Request) {
 	_, err := rc.getCredentials(r)
 	// if a token is not found, return Forbidden.
 	if authErrors.IsTokenNotFound(err) {
-		rc.forbiddenResponse(w, err)
+		rc.forbidden(w, err)
+		return
+	}
+
+	rc.deleteSessionCookie(w, rc.configuration.AccessTokenCookie)
+	w.WriteHeader(http.StatusOK)
+}
+
+// GetToken returns data from the Vault.
+func (rc RESTController) GetToken(c *gin.Context) {
+	r, w := c.Request, c.Writer
+	resource := strings.Split(r.URL.Path, "/")[2]
+	uuid := c.Params.ByName("uuid")
+	path := fmt.Sprintf("%s/data/%s", resource, uuid)
+
+	data, err := rc.provisionTokenService.Get(path)
+	if err != nil {
+		rc.badRequest(w, err)
+		return
+	}
+
+	json.NewEncoder(w).Encode(data)
+	w.WriteHeader(http.StatusOK)
+}
+
+// ListTokens returns a data list from the Vault.
+func (rc RESTController) ListTokens(c *gin.Context) {
+	r, w := c.Request, c.Writer
+	resource := strings.Split(r.URL.Path, "/")[2]
+	uuid := c.Params.ByName("uuid")
+	path := fmt.Sprintf("%s/data/%s", resource, uuid)
+
+	data, err := rc.provisionTokenService.List(path)
+	if err != nil {
+		rc.badRequest(w, err)
+		return
+	}
+
+	json.NewEncoder(w).Encode(data)
+	w.WriteHeader(http.StatusOK)
+}
+
+// CreateToken stores data into Vault.
+func (rc RESTController) CreateToken(c *gin.Context) {
+	r, w := c.Request, c.Writer
+
+	var newProvisionToken models.ProvisionTokenRequest
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		rc.internalServerError(w, err)
+		return
+	}
+	json.Unmarshal(reqBody, &newProvisionToken)
+
+	resource := strings.Split(r.URL.Path, "/")[2]
+	uuid := c.Params.ByName("uuid")
+	path := fmt.Sprintf("%s/data/%s", resource, uuid)
+
+	data, err := rc.provisionTokenService.Create(path, newProvisionToken)
+	if err != nil {
+		rc.badRequest(w, err)
+		return
+	}
+
+	expiration := time.Duration(newProvisionToken.Expiration) * time.Hour
+	jsonBody, err := rc.oidcService.Create(data, expiration)
+	if err != nil {
+		rc.internalServerError(w, err)
 		return
 	}
 
-	rc.deleteSessionCookie(w, "access_token")
+	json.NewEncoder(w).Encode(jsonBody)
+	w.WriteHeader(http.StatusCreated)
+}
+
+// DeleteToken removes a data from the Vault.
+func (rc RESTController) DeleteToken(c *gin.Context) {
+	r, w := c.Request, c.Writer
+	resource := strings.Split(r.URL.Path, "/")[2]
+	uuid := c.Params.ByName("uuid")
+	path := fmt.Sprintf("%s/data/%s", resource, uuid)
+
+	tokenToBeDeleted := c.Query("token_id")
```

Comment:
we spoke about this in the slack 😅  let's keep with the original methods

Reply:
Let's use path parameters and define clear paths for each resource/method.

```
GET /profiles/profile-id/tokens
POST /profiles/profile-id/tokens
DELETE /profiles/profile-id/tokens/token-id

GET /synthesizers/synthesizer-id/tokens
POST /synthesizers/synthesizer-id/tokens
DELETE /synthesizers/synthesizer-id/tokens/token-id
```

Reply:
agreeeee